src/magnify/apps/core/models.py

```diff
@@ -235,39 +235,49 @@ def jitsi_name(self):
         """The name used for the room in Jitsi."""
         return f"{self.slug:s}-{self.id!s}"
 
-    def is_administrator(self, user):
+    def get_role(self, user, direct_only=False):
         """
-        Check if a user is administrator of the room.
+        Determine the role of a given user in this room.
 
-        Users carrying the "owner" role are considered as administrators a fortiori.
+        direct_only: boolean
+            Whether or not to take into account the roles a user inherits from
+            its membership in a group.
         """
         if not user or not user.is_authenticated:
-            return False
+            return None
 
-        return (
-            self.user_accesses.filter(
-                role__in=[UserRoleChoices.ADMIN, UserRoleChoices.OWNER], user=user
-            ).exists()
-            or self.group_accesses.filter(
+        try:
+            # pylint: disable=no-member
+            role = self.user_accesses.get(user=user).role
+        except self.user_accesses.model.DoesNotExist:
+            role = None
+
+        if direct_only:
+            return role
+
+        if not (role and role != UserRoleChoices.MEMBER):
+            for access in self.group_accesses.filter(
```

Comment:
```suggestion
       if direct_only or role not in (None, UserRoleChoices.MEMBER): :
            return role

        for access in self.group_accesses.filter(
```
I find that easier to read

Reply:
Good idea! Very elegant :pray:

Reply:
fix: https://github.com/openfun/jitsi-magnify/pull/135/commits/b34bd7e9a6ab2c59ee54bb7f08bec385d4500037